src/smt/theory_str_noodler/state_len.h

```diff
@@ -121,11 +109,11 @@ namespace smt::noodler {
             m_util_a(util_a) 
             { }
 
-        void initialize(const Instance& inst) override {
+        void initialize(const Instance& inst) {
             this->state.add(inst, 0);
         }
 
-        bool get_another_solution(const Instance& inst, LengthConstr& out) override {
+        bool get_another_solution(const Instance& inst, LengthConstr& out) {
             int cnt = this->state.get_val(inst);
             if(cnt >= 10) {
                 return false;
```

Comment:
These functions can be left as they are. Eventually, we will rewrite this debug decision procedure if we need to debug something later on.

The implemented functions from `AbstractDecisionProcedure` should probably fail on assert or throw en exception with text "unimplemented" so that it compiler while inheriting from `AbstractDecisionProcedure`, but we want to know about that and if our tests start failing on this assert/exception, modify the debug decision procedure.